pkg/validation-webhook/snapshot_test.go

```diff
@@ -644,3 +663,614 @@ func TestAdmitVolumeSnapshotClassV1(t *testing.T) {
+				Spec: volumegroupsnapshotv1alpha1.VolumeGroupSnapshotContentSpec{
+					Source: volumegroupsnapshotv1alpha1.VolumeGroupSnapshotContentSource{
+						VolumeGroupSnapshotHandle: &modifiedField,
+					},
+					VolumeGroupSnapshotRef: core_v1.ObjectReference{
+						Name:      "snapshot-ref",
+						Namespace: "default-ns",
+					},
+				},
+			},
+			oldGroupSnapContent: validContent,
+			shouldAdmit:         false,
+			operation:           v1.Update,
+			msg:                 fmt.Sprintf("Spec.Source.VolumeGroupSnapshotHandle is immutable but was changed from %s to %s", groupSnapshotHandle, modifiedField),
+		},
+		{
+			name:                "Update: old is invalid and new is valid",
+			groupSnapContent:    validContent,
+			oldGroupSnapContent: invalidContent,
+			shouldAdmit:         false,
+			operation:           v1.Update,
+			msg:                 fmt.Sprintf("Spec.Source.PersistentVolumeNames is immutable but was changed from %s to %s", []string{volumeHandle}, []string{}),
+		},
+		{
+			name:                "Update: old is invalid and new is invalid",
+			groupSnapContent:    invalidContent,
+			oldGroupSnapContent: invalidContent,
+			shouldAdmit:         false,
+			operation:           v1.Update,
+			msg:                 "both Spec.VolumeGroupSnapshotRef.Name =  and Spec.VolumeGroupSnapshotRef.Namespace = default-ns must be set",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			groupSnapContent := tc.groupSnapContent
+			raw, err := json.Marshal(groupSnapContent)
+			if err != nil {
+				t.Fatal(err)
+			}
+			oldGroupSnapContent := tc.oldGroupSnapContent
+			oldRaw, err := json.Marshal(oldGroupSnapContent)
+			if err != nil {
+				t.Fatal(err)
+			}
+			review := v1.AdmissionReview{
+				Request: &v1.AdmissionRequest{
+					Object: runtime.RawExtension{
+						Raw: raw,
+					},
+					OldObject: runtime.RawExtension{
+						Raw: oldRaw,
+					},
+					Resource:  GroupSnapshotContentV1Apha1GVR,
+					Operation: tc.operation,
+				},
+			}
+			sa := NewGroupSnapshotAdmitter(nil)
+			response := sa.Admit(review)
+			shouldAdmit := response.Allowed
+			msg := response.Result.Message
+
+			expectedResponse := tc.shouldAdmit
+			expectedMsg := tc.msg
+
+			if shouldAdmit != expectedResponse {
+				t.Errorf("expected \"%v\" to equal \"%v\"", shouldAdmit, expectedResponse)
+			}
+			if msg != expectedMsg {
+				t.Errorf("expected \"%v\" to equal \"%v\"", msg, expectedMsg)
+			}
+		})
+	}
+}
+
+func TestAdmitVolumeGroupSnapshotClassV1Alpha1(t *testing.T) {
+	testCases := []struct {
+		name              string
+		groupSnapClass    *volumegroupsnapshotv1alpha1.VolumeGroupSnapshotClass
+		oldGroupSnapClass *volumegroupsnapshotv1alpha1.VolumeGroupSnapshotClass
+		shouldAdmit       bool
+		msg               string
+		operation         v1.Operation
+		lister            groupSnapshotListers.VolumeGroupSnapshotClassLister
+	}{
+		{
+			name: "new default for class with no existing classes",
```

Comment:
new default for class with no existing classes -> new default for group snapshot class with no existing group snapshot classes"